File: webApp/src/app/services/api/pets/pets.service.ts
import { Picture } from './../../../shared/models/dataModels';
import { Injectable } from '@angular/core';
import { catchError } from 'rxjs/internal/operators/catchError';
import { Pet } from 'src/app/shared/models/dataModels';
import { DataService } from '../data.service';

@Injectable({
  providedIn: 'root'
})
export class PetsService extends DataService {


  getPets() {
    return this.http.get<Pet[]>(`${this.url}/pets`).pipe(catchError(this.handleError))
  }

  addPets(pet: Pet) {
    return this.http.post<Pet>(`${this.url}/pets`, pet).pipe(catchError(this.handleError))
  }

  deletePet(pet: Pet) {
    console.log(pet.id)
    return this.http.delete(`${this.url}/pets/${pet.id}`).pipe(catchError(this.handleError))
  }

  updatePet(pet: Pet) {
    return this.http.put<Pet>(`${this.url}/pets/${pet.id}`, pet).pipe(catchError(this.handleError))

  }

}

File: mobile/lib/services/api/pets/pets_service.dart
import 'dart:convert';

import 'package:dio/dio.dart';
import 'package:injectable/injectable.dart';
import 'package:PetRoulette/app_config.dart';
import 'package:PetRoulette/models/models.dart';
import 'package:PetRoulette/services/data_service.dart';

@injectable
class PetsService extends DataService {
  PetsService(AppConfig config) : super(config);

  Future<List<Pet>> getPets() async {
    Response response = await dio.get(apiUrl + "/pets");
    final json = response.data as List<Object>;

    return json.map((e) => Pet.fromJson(e)).toList();
  }

  Future<Pet> addPets(Pet pet) async {
    Response response = await dio.post(apiUrl + "/pets", data: pet);

    return Pet.fromJson(response.data);
  }

  Future<Pet> updatePet(Pet pet) async {
    Response response = await dio.put(apiUrl + "/pets/${pet.id}", data: pet);

    return Pet.fromJson(response.data);
  }

  Future<void> deletePet(Pet pet) async {
    await dio.delete(apiUrl + "/pets/${pet.id}");
  }
}
